internal/gengapic/genrest.go

```diff
@@ -131,24 +230,39 @@ func (g *generator) unaryRESTCall(servName string, m *descriptor.MethodDescripto
 		lowcaseServName, m.GetName(), inSpec.Name, inType.GetName(), outSpec.Name, outType.GetName())
 
 	// TODO(dovs): handle cancellation, metadata, osv.
-	// TODO(dovs): handle request from http body opt
-	p("m := jsonpb.Marshaler{}") // TODO(dovs): add appropriate options
-	p("if jsonReq, err := m.MarshalToString(req); err != nil {")
+	// TODO(dovs): handle http headers
+	// TODO(dovs): handle deadlines?
+	// TODO(dovs): handle call options
+	// TODO(dovs): handle path parameters
+	p("// The default (false) for the other options are fine.")
+	p("marshaler := protojson.MarshalOptions{AllowPartial: true, Multiline: true, EmitUnpopulated: true}")
+	p("if jsonReq, err := marshaler.Marshal(req); err != nil {")
+	p("  return nil, err")
+	p("}")
+	p("")
+	p("url := fmt.Sprintf(\"https://%%s%s\", c.host)", info.url)
+	p("httpReq, err := http.NewRequestWithContext(ctx, \"%s\", url, bytes.NewReader(jsonReq))", strings.ToUpper(info.verb))
+	p("if err != nil {")
 	p("    return nil, err")
+	p("}")
 	p("")
+	p("httpRsp, err := client.Do(httpReq)")
+	p("if err != nil{")
+	p(" return nil, err")
+	p("} else if httpRsp.StatusCode >= 400 {")
+	p("  return nil, errors.E(nil, httpRsp.Status)")
 	p("}")
-	// TODO(dovs): add real method logic.
-	// _ := http_opt(m)
-
-	// * JSONify the request
-	// * determine the url
-	// ** Handle query parameters
-	// * send the request, receive the response, maybe return an error
-	// * deserialize the result and return
-
-	p("return nil, nil")
 	p("")
+	p("buf := new(bytes.Buffer)")
+	p("err = buf.ReadFrom(httpRsp.Body)")
```

Comment:
Does `buf` auto-close the Body? If not, do we need to `defer` that?

Reply:
It is something we need to do. Good catch.